### ISSUE TITLE
Play 2.4 encourages removing GlobalSettings.onStart, refactor example

### ISSUE THREAD

**Issue by Ciantic** [playframework/play-slick#279]
I read about GlobalSettings deprecation from [Play 2.4 migration guide](https://www.playframework.com/documentation/2.4.x/GlobalSettings). 

The computer-database example needs to be refactored, it contains initial data in onStart:
https://github.com/playframework/play-slick/blob/master/samples/computer-database/app/Global.scala

Thing is I don't know how to implement initial data even with di example, it doesn't include initial data.

Perhaps it's required to inherit from `DatabaseConfigProvider` as `MyApplicationDatabaseConfigProvider` and create initial data in that constructor, and pass it with DI to all DAO classes?

**Edit** Also added [a question in Stack Overflow about DI and database initialization.](http://stackoverflow.com/questions/30953517/how-to-do-database-initialization-when-using-di-in-play-2-4)

**Comment by iref** [comment]
I solved initial data creation recently using very similar approach as bjfletcher suggested in his SO answer.
I implemented `DatabaseSeed` class (probably not very good name), which has dependency on DAO classes, that are required to insert data. It runs data creation during construction and it's registered in custom Guice module as eager singleton.

This approach is also suggested in Play 2.4 migration guide, section about refactoring `GlobalSettings.onStart`. I can try to refactor computer-database example using this approach and submit PR if @dotta and other play-slick commiters agree that this is way to go in 2.4
